server/engine/chainer.go

```diff
@@ -144,22 +159,23 @@ func (m *ChainEngine) Remove(p string) error {
 // Each operation (up and down) contains a
 // - Error channel gives all the errors along the way until it closes
 // - Status channel gives all the message status along the way
-func (m *ChainEngine) Run(p string, c *ChainerContext, upOp Operation, downOp Operation, done chan bool) {
+func (m *ChainEngine) Run(p string, c *ChainerContext, notifier StepNotifier, done chan bool) {
 
 	defer func() {
```

Comment:
dans ce cas le range est le plus adapté de toute manière mais c'est bizare ce que tu me dis parce que j'ai fait un for avec select dedans pour le taff vu que j'avais deux channels à écouter et ça marche très bien.

En tout cas c'est vachement plus propre et clair comme ça :+1: